blocks/handler.go

```diff
@@ -60,9 +61,13 @@ func (h *Handler) HandleSyncedBlock(ctx context.Context, data []byte) error {
 		logger.With().Error("malformed block", log.Err(err))
 		return errMalformedData
 	}
-
 	// set the block ID when received
 	b.Initialize()
+
+	if err := vm.ValidateRewards(b.Rewards); err != nil {
```

Comment:
why do it here instead of after the de-duplication check?

Reply:
because syntactic validation is cheap. i can move it afterwards, doesn't matter much to me